src/adapters/RoomsJSONAdapter.js

```diff
@@ -0,0 +1,82 @@
+import {Observable} from 'rxjs';
+
+import RoomsAdapter from './RoomsAdapter';
+
+/**
+ * @typedef RoomJSON
+ * @param {object} datasource An object that contains a set of rooms keyed by ID.
+ * @example
+ * {
+ *  "room-1": {
+ *     "ID": "room-1",
+ *     "title": "title",
+ *     "roomType": "group"
+ *  },
+ *  "room-1-activities": []
+ * }
+ */
+
+/*
+ * Implements the RoomsAdapter interface with a JSON object as its datasource. See @RoomsJSON
+ */
+export default class RoomsJSONAdapter extends RoomsAdapter {
+  constructor(datasource) {
+    super(datasource);
+    this.getRoom = this.getRoom.bind(this);
+    this.getPreviousRoomActivities = this.getPreviousRoomActivities.bind(this);
+    this.getRoomActivities = this.getRoomActivities.bind(this);
+  }
+
+  /**
+   * Returns an observable that emits room data of the given ID.
+   *
+   * @param {string} ID ID of room to get
+   * @returns {Observable.<Room>}
+   * @memberof RoomsJSONAdapter
+   */
+  getRoom(ID) {
+    return Observable.create((observer) => {
+      if (this.datasource[ID]) {
+        observer.next(this.datasource[ID]);
+      } else {
+        observer.error(new Error(`Could not find room with ID "${ID}"`));
+      }
+
+      observer.complete();
+    });
+  }
+
+  /**
+   * Returns an observable that emits an array of previous activity IDs of the given roomID.
+   *
+   * @param {string} ID  ID of the room to get.
+   * @returns {Observable.<Array.<string>>}
+   * @memberof RoomsAdapter
+   */
+  getPreviousRoomActivities(ID) {
+    return Observable.create((observer) => {
+      const data = !this.datasource[ID] ? [] : this.datasource[ID];
+
+      observer.next(data);
+
+      observer.complete();
+    });
+  }
+
+  /**
+   * Returns an observable that emits an array of current activity IDs of the given roomID.
+   *
+   * @param {string} ID  ID of the room to get.
+   * @returns {Observable.<Array.<string>>}
+   * @memberof RoomsAdapter
+   */
+  getRoomActivities(ID) {
+    return Observable.create((observer) => {
+      const data = !this.datasource[ID] ? [] : this.datasource[ID];
```

Comment:
If a user passes in the room ID as defined in the jsdoc, this will not return the activities array, this returns the room data. 

It should do: 
```js
this.datasource[`${ID}-activities`]
```

This way we can still just use the room ID and don't have to pass the knowledge of adding a `-activities`, especially since the RoomsAdapter defines the prop as the room ID and not room ID + activities string.

Reply:
yeah I was going to do that at the hook level, but adapter should be fine too.